cli/listener/src/main.rs

```diff
@@ -392,7 +405,170 @@ impl Metrics {
+    let response_interval = ResponseInterval {
+        annual_percentage_yield: interval_prices.annual_percentage_yield(),
+        interval_prices,
+    };
+    let content_type = Header::from_bytes(
+        &b"Content-Type"[..],
+        &b"application/json; charset=UTF-8"[..],
+    )
+    .expect("Static header value, does not fail at runtime.");
+    Response::from_data(
+        serde_json::to_vec(&response_interval).expect("Serialization shouldn't fail"),
+    )
+    .with_header(content_type)
+    .boxed()
+}
+
+/// Gets a response that will be sent to a requesting client for a price
+/// interval. Returns None if no method is passed.
+fn get_interval_price_request(
+    db_connection: &Connection,
+    request_url: &str,
+) -> Option<ResponseBox> {
+    let parsed_url = match Url::parse(request_url) {
+        Ok(parsed_url) => parsed_url,
+        Err(_err) => {
+            return Some(get_error_response(ResponseError::new_bad_request(
+                "Error parsing Url",
+            )));
+        }
+    };
+    let method_name = parsed_url.path_segments()?.last()?;
+    if method_name != "apy" {
+        return Some(get_error_response(ResponseError::new_bad_request(
+            "Method not supported, use \"/apy?begin=<begin_date_iso8601>&end=<end_date_iso8601>\"",
+        )));
+    }
+    let parsed_request_url =
+        form_urlencoded::parse(parsed_url.query()?.as_bytes()).collect::<Vec<_>>();
+
+    let dates = match get_date_params(parsed_request_url) {
+        Ok(res) => Some(res),
+        Err(err) => return Some(get_error_response(err)),
+    }?;
+
+    let interval_prices = get_interval_price_for_period(
+        db_connection
+            .unchecked_transaction()
+            .expect("Failed to create sqlite transaction."),
+        dates.start,
+        dates.end,
+        SOLIDO_ID.to_owned(),
+    );
+    match interval_prices {
+        // Error while getting the interval prices.
+        Err(err) => {
+            eprintln!("Internal Error when getting interval prices: {}", err);
+            Some(get_error_response(ResponseError::InternalServerError))
+        }
+        Ok(interval_prices_opt) => {
+            if let Some(interval_prices) = interval_prices_opt {
+                // Got interval prices.
+                Some(get_success_response(interval_prices))
+            } else {
+                // No interval price could be calculated, probably because of few data points.
+                Some(get_error_response(ResponseError::new_bad_request(
+                    "No data points for calculating the price interval.",
+                )))
+            }
+        }
+    }
+}
+
+fn serve_request(
+    db_connection: &Connection,
+    request: Request,
+    metrics_mutex: &MetricsMutex,
+) -> Result<(), std::io::Error> {
+    if let Some(res) = get_interval_price_request(db_connection, request.url()) {
+        return request.respond(res);
+    };
```

Comment:
Hmm, so now we serve the APY response on any url that ends with `/apy`, and the metrics on anything else. I think it would be safer to turn that around, and serve the metrics only on `/metrics`.